db/db_impl.h

```diff
@@ -848,14 +851,29 @@ class DBImpl : public DB {
                             bool disable_memtable = false,
                             uint64_t* seq_used = nullptr);
 
-  // batch_cnt is expected to be non-zero in seq_per_batch mode and indicates
-  // the number of sub-patches. A sub-patch is a subset of the write batch that
-  // does not have duplicate keys.
-  Status WriteImplWALOnly(const WriteOptions& options, WriteBatch* updates,
-                          WriteCallback* callback = nullptr,
-                          uint64_t* log_used = nullptr, uint64_t log_ref = 0,
-                          uint64_t* seq_used = nullptr, size_t batch_cnt = 0,
-                          PreReleaseCallback* pre_release_callback = nullptr);
+  // Write only to memtables without joining any write queue
+  Status UnorderedWriteMemtable(const WriteOptions& write_options,
+                                WriteBatch* my_batch, WriteCallback* callback,
+                                uint64_t log_ref, uint64_t seq);
+
+  // Whether the batch requires to be assigned with an order
+  enum AssignOrder { DontAssignOrder, DoAssignOrder };
+  // Whether it requires publishing last sequence or not
+  enum PublishLastSeq { DontPublishLastSeq, DoPublishLastSeq };
+
+  // sub_batch_cnt is expected to be non-zero when assign_order = DoAssignOrder
+  // indicating the number of sub-batches in my_batch. A sub-patch is a subset
+  // of the write batch that does not have duplicate keys. When seq_per_batch is
+  // not set, each key is a separate sub_batch. Otherwise each duplicate key
+  // marks start of a new sub-batch.
+  Status WriteImplWALOnly(WriteThread& write_thread,
```

Comment:
This function is for write requests where memtable write is skipped. Now we re-purpose it to include the WAL writing step too. At least the function name should be changed.